test/performance/search/index_construction_benchmark.cpp

```diff
@@ -0,0 +1,196 @@
+    }
+}
+
+#if SEQAN3_HAS_SEQAN2
+struct sequence_store_seqan2
+{
+    seqan::String<seqan::Dna> const dna4_rng{seqan3::test::generate_sequence_seqan2<seqan::Dna>(max_length, 0, seed)};
+    seqan::String<seqan::AminoAcid> const aa27_rng{seqan3::test::generate_sequence_seqan2<seqan::AminoAcid>(max_length, 0, seed)};
+    seqan::String<char> const char_rng{seqan3::test::generate_numeric_sequence<uint8_t>(max_length, 0, 253, seed)
+                                       | seqan3::views::persist
+                                       | seqan3::views::rank_to<char>
+                                       | seqan3::views::to<std::string>};
+};
+
+sequence_store_seqan2 store2;
+
+// Since the seqan2 alphabet has a value type, the dimension is actually one less than seqan3::dimension_v reports.
+template <typename rng_t>
+constexpr size_t seqan_dimension_v()
+{
+    if (seqan3::detail::is_type_specialisation_of_v<rng_t, seqan::String>)
+        return 1;
+    if (seqan3::detail::is_type_specialisation_of_v<rng_t, seqan::StringSet>)
+        return 2;
+   return 0;
+}
+
+template <tag index_tag, typename rng_t>
+void index_benchmark_seqan2(benchmark::State & state)
+{
+    constexpr size_t dimension = seqan_dimension_v<rng_t>();
+    static_assert(dimension != 0, "Use seqan::String or seqan::StringSet for SeqAn2 index benchmarks!");
+
+    // Calling std::ranges::range_value_t twice on seqan::String<char> is not valid.
+    using alphabet_t = seqan3::detail::lazy_conditional_t<dimension == 1,
+                                                         std::ranges::range_value_t<rng_t>,
+                                                         seqan3::detail::lazy<std::ranges::range_value_t, std::ranges::range_value_t<rng_t>>>;
+
+    using inner_rng_t = std::conditional_t<dimension == 1, rng_t, std::ranges::range_value_t<rng_t>>;
+    using index_cfg = seqan::FastFMIndexConfig<void, uint32_t>;
+
+    rng_t sequence;
+    inner_rng_t inner_sequence;
+
+    if constexpr (std::same_as<alphabet_t, seqan::Dna>)
+        inner_sequence = seqan::prefix(store2.dna4_rng, state.range(0));
+    else if constexpr (std::same_as<alphabet_t, seqan::AminoAcid>)
+        inner_sequence = seqan::prefix(store2.aa27_rng, state.range(0));
+    else
+        inner_sequence = store2.char_rng;
+
+    if constexpr (dimension == 1)
+        sequence = std::move(inner_sequence);
+    else
+        for (int32_t i = 0; i < state.range(1); ++i)
+            seqan::appendValue(sequence, inner_sequence);
+
+    for (auto _ : state)
+    {
+        if constexpr (index_tag == tag::fm_index)
+        {
+            seqan::Index<rng_t, seqan::FMIndex<void, index_cfg>> index{sequence};
+            seqan::indexCreate(index, seqan::FibreSALF());
+        }
+        else
+        {
+            seqan::Index<rng_t, seqan::BidirectionalIndex<seqan::FMIndex<void, index_cfg>>> index{sequence};
+            seqan::indexCreate(index, seqan::FibreSALF());
+        }
```

Comment:
```cpp
using index_type = std::conditional_t<index_tag == tag::fm_index, seqan::FMIndex<void, index_cfg>, seqan::BidirectionalIndex<seqan::FMIndex<void, index_cfg>>>;
seqan::Index<rng_t, index_type> index{sequence};
seqan::indexCreate(index, seqan::FibreSALF());
```
instead of the if else clause